### ISSUE TITLE
Bug: Cannot encode Timestamp - When you create a change with a Timestamp value

### ISSUE THREAD

**Issue by dauledk** [firebase/firebase-functions-test#17]
### Version info

**firebase-functions-test:**
0.1.2
**firebase-functions:**
1.0.2
**firebase-admin:**
5.12.1

### Test case

Create a change object, from a object that contains a Timestamp value
```
const afterSnap = test.firestore.makeDocumentSnapshot(mockTask, 'some-path');
const change = test.makeChange(null, afterSnap);
```

### Expected behavior

Should works as expected, since firestore is changing to Timestamp instead of js-date

### Actual behavior

Throws this error:

 "Cannot encode Timestamp(seconds=1515157200, nanoseconds=0)to a Firestore Value. Local testing does not yet support Firestore document reference values or geo points."

**Comment by laurenzlong** [comment]
Thanks for filing! How did you construct "mockTask"?

**Comment by dauledk** [comment]
I ported the code from here: https://github.com/firebase/firebase-js-sdk/blob/1608edb5f1b8d3dae342727b4914e3edc8b5dcd8/packages/firestore/src/api/timestamp.ts

And thus, I just created a timestamp with `const timestamp = Timestamp.fromDate(new Date())`

And some background info:
Currently (to my knowledge) it is not possible to create a Timestamp object with firebase-admin. But since our team enforces code sharing between our client repo and functions repo, (and write code in typescript and hence uses interfaces), we need to create Timestamps in our func repo.

**Comment by laurenzlong** [comment]
Thanks for the clarification. There will be an update to the Node SDK for Firestore that will soon incorporate the new changes (from JS date to timetamp). When that happens, I will also update this library to allow timestamps to be passed into makeDocumentSnapshot.

**Comment by gustavopch** [comment]
@laurenzlong Any news?

**Comment by mattvukas** [comment]
Also interested in this, would be very useful for testing Cloud Function business logic around document dates changing.

**Comment by VladimirUA** [comment]
Face to same provlem. Any updates?

**Comment by abehaskins** [comment]
This issue was resolved with #28.
